src/write_gcm.c

```diff
@@ -2513,19 +2529,33 @@ static wg_context_t *wg_context_create(const wg_configbuilder_t *cb) {
     goto leave;
   }
 
-  const char *format_string_to_use =
+  const char *ats_format_string_to_use =
       cb->agent_translation_service_format_string != NULL ?
           cb->agent_translation_service_format_string :
           agent_translation_service_default_format_string;
 
-  char url[512];  // Big enough?
-  int sprintf_result = snprintf(url, sizeof(url), format_string_to_use,
+  char ats_url[512];  // Big enough?
+  int sprintf_result = snprintf(ats_url, sizeof(ats_url), ats_format_string_to_use,
       build->resource->project_id);
-  if (sprintf_result < 0 || sprintf_result >= sizeof(url)) {
+  if (sprintf_result < 0 || sprintf_result >= sizeof(ats_url)) {
     ERROR("write_gcm: overflowed url buffer");
     goto leave;
   }
-  build->agent_translation_service_url = sstrdup(url);
+  build->agent_translation_service_url = sstrdup(ats_url);
+
+  const char *cm_format_string_to_use =
+    cb->custom_metrics_format_string != NULL ?
+    cb->custom_metrics_format_string :
+    custom_metrics_default_format_string;
+
+  char cm_url[512];  // Big enough?
```

Comment:
The metric type will never be part of this URL, only the project id. I've added some checks to ensure that the project id fits within the length limit, and pushed the new code.